### DISCUSSION TITLE
The MediaPlayerElement in GTK can't play media.

### DISCUSSION THREAD

**Discussion by LionelXN** [unoplatform/uno#12944] [Q&A]
Hi, on Uno version 4.9.17, I used MediaPlayerElement in the GTK project and it couldn't play properly on Linux. What is the reason?
![image](https://github.com/unoplatform/uno/assets/18021306/9eb1404f-fbab-4463-9f4c-a7a4527f8f06)

![image](https://github.com/unoplatform/uno/assets/18021306/37dd565a-66eb-407e-88f4-badb5cc278bb)

![f5c3d341555634223312047fd3aab123](https://github.com/unoplatform/uno/assets/18021306/82bfe3ea-ca38-4b03-845d-74e4423e3432)

![image](https://github.com/unoplatform/uno/assets/18021306/4f45b7bb-34c4-412f-8610-4b03aa93f0c7)

**Comment by jeromelaban** [comment]
Thanks for the report. This is available in uno preview builds, since https://github.com/unoplatform/uno/pull/12606.

**Reply by Sharktao** [reply]
![lQLPJv_D0ml_ysLMj80CDrBgvZ8p5iXLAgSvGKZFwJoA_526_143](https://github.com/unoplatform/uno/assets/43270517/b4833eed-b3a6-403d-b220-7c3a85a60fb8)
![lQLPJxbScUoXasJOzQNssJeQdnAiC4ZmBK8YpkQAaAA_876_78](https://github.com/unoplatform/uno/assets/43270517/2f14eed0-158e-457f-bf45-6a864ab0c967)
HI,I upgraded to the latest, as shown in the picture running error

**Reply by jeromelaban** [reply]
Thanks. It seems that libvlc is not handling something properly. Could you [enable logging](https://platform.uno/docs/articles/logging.html), so we can see further what my be the issue? You may want to enable debug logging for the `Uno.UI.Media` namespace.

**Reply by Sharktao** [reply]
![419E4918-0C95-469b-8282-6FD837571D41](https://github.com/unoplatform/uno/assets/43270517/2a9776a6-0f1e-48b7-a859-b27cc5a6bebd)

Will it help?

**Reply by jeromelaban** [reply]
Thanks for the update. So given the code you're showing, it would seem that you're starting the player immediately up the app startup. I wonder if this is related. Could you create a separate page, or disable autoplay to see if this helps (so that the playback starts later).

**Reply by Sharktao** [reply]
![5DB83CE2-7AEB-45b1-8A3D-FF74CE15DBAC](https://github.com/unoplatform/uno/assets/43270517/124d743f-b05a-44b2-b530-cad5ad084b8e)
![D632CE98-829C-48c4-84C5-70AD6FD23415](https://github.com/unoplatform/uno/assets/43270517/982f411d-3ca5-4499-b572-c5acd7de02ec)
I set the property，  AutoPlay = false；

**Reply by jeromelaban** [reply]
Thanks for the update. This may look like a platform specific issue that we have not encountered. Are you able to use GDB to get a stack trace of the segfault?

**Reply by jeromelaban** [reply]
> Sorry, due to some reasons, I can not go further, I made the attempt as shown in the figure, run in the offer, but the prompt can not find the file path, attached is the test code.

I'm not sure to understand your sentence. Did you mean that you attached a file with the sample to test? Also, to be able to debug further, can you provide a GDB stack trace? We won't be otherwise able to tell what's causing the crash.

**Comment by Sharktao** [comment]
Sorry, due to some reasons, I can not go further, I made the attempt as shown in the figure, run in the offer, but the prompt can not find the file path, attached is the test code.



***@***.***
 
From: Jérôme Laban
Date: 2023-07-25 20:59
To: unoplatform/uno
CC: Sharktao; Comment
Subject: Re: [unoplatform/uno] The MediaPlayerElement in GTK can't play media. (Discussion #12944)
Thanks for the update. This may look like a platform specific issue that we have not encountered. Are you able to use GDB to get a stack trace of the segfault?

**Comment by Sharktao** [comment]
My job has changed；  
        The video feature was shelved。



***@***.***
 
From: Jérôme Laban
Date: 2023-07-26 19:45
To: unoplatform/uno
CC: Sharktao; Comment
Subject: Re: [unoplatform/uno] The MediaPlayerElement in GTK can't play media. (Discussion #12944)
Sorry, due to some reasons, I can not go further, I made the attempt as shown in the figure, run in the offer, but the prompt can not find the file path, attached is the test code.
I'm not sure to understand your sentence. Did you mean that you attached a file with the sample to test? Also, to be able to debug further, can you provide a GDB stack trace? We won't be otherwise able to tell what's causing the crash.
